pkg/porter/editor.go

```diff
@@ -0,0 +1,86 @@
+package porter
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"get.porter.sh/porter/pkg/context"
+)
+
+const (
+	defaultEditor        = "vi"
+	defaultEditorWindows = "notepad"
+)
+
+// Editor displays content to a user using an external text editor, like vi or notepad.
+// The content is captured and returned.
+//
+// The `EDITOR` environment variable is checked to find an editor.
+// Failing that, use some sensible default depending on the operating system.
+//
+// This is useful for editing things like configuration files, especially those
+// that might be stored on a remote server. For example: the content could be retrieved
+// from the remote store, edited locally, then saved back.
+type Editor struct {
+	Data     []byte
+	Filename string
+	Context  *context.Context
```

Comment:
The pattern we use elsewhere is to embed the context struct so that we can access its members without having to do `x.Context.FileSystem` or `x.Context.Out`, and can instead do `x.Filesystem` and `x.Out`, `x.Debug`, etc. So let's change this to be consistent.

```suggestion
	*context.Context
```

Reply:
Sounds good, will do that!